test/error/error.test.js

```diff
@@ -1,145 +1,83 @@
 'use strict'
 
-var DOMParser = require('../../lib/dom-parser').DOMParser
-const assert = require('../assert')
+const { DOMParser } = require('../../lib/dom-parser')
 
-var xml =
-	'<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0"\n\
-       profile="ecmascript" id="scxmlRoot" initial="start">\n\
-\n\
-  <!--\n\
-      some comment (next line is empty)\n\
-\n\
-  -->\n\
-\n\
-  <state id="start" name="start">\n\
-    <transition event"init" name="init" target="main_state" />\n\
-  </state>\n\
-\n\
-  </scxml>'
-var error = []
-var parser = new DOMParser({
-	locator: {},
-	errorHandler: {
-		error: function (msg) {
-			error.push(msg)
-		},
-	},
-})
-var doc = parser.parseFromString(xml, 'text/html')
-var doc = parser.parseFromString(
-	'<html><body title="1<2"><table>&lt;;test</body></body></html',
-	'text/html'
-)
+const XML_ERROR = '<html><body title="1<2"><table&lt;;test</body></body></html>'
+const XML_ERROR_AND_WARNING = '<html disabled><1 1="2"/></body></html>'
 
-describe('errorHandle', () => {
+describe('errorHandler', () => {
 	it('only single function with two args builds map', () => {
-		var error = {}
-		var parser = new DOMParser({
+		const errors = {}
+		const parser = new DOMParser({
 			errorHandler: function (key, msg) {
-				error[key] = msg
+				errors[key] = msg
 			},
 		})
-		var doc = parser.parseFromString(
-			'<html disabled><1 1="2"/></body></html>',
-			'text/xml'
-		)
-		assert.isTrue(
-			typeof error.warning === 'string',
-			'error.warning: ' + String(error.warning)
-		)
-		assert.isTrue(
-			typeof error.error === 'string',
-			'error.error: ' + String(error.error)
-		)
-	})
 
-	it('only one function with one argument builds list', () => {
-		var error = []
-		var parser = new DOMParser({
-			errorHandler: function (msg) {
-				error.push(msg)
-			},
-		})
-		var doc = parser.parseFromString(
-			'<html disabled><1 1="2"/></body></html>',
-			'text/xml'
-		)
-		error.map(function (e) {
-			error[/^\[xmldom (\w+)\]/.exec(e)[1]] = e
+		parser.parseFromString(XML_ERROR_AND_WARNING, 'text/xml')
```

Comment:
Yes, but I think it's as simple as it can get.

We could of course introduce a small module that holds the most common mime types, or at least the ones we test for. Or even add some devDependency for that, but is it really worth it?

If you think it is I would prefer to do that change in a separate PR.

Reply:
OK. I would love to avoid hard-coded strings, but this is probably not worth changing now.